[quant][graphmode] Add quantization support for aten::cat

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35130 [quant][graphmode] Make interpolate/upsample work again
* #35142 [quant][graphmode] SwapDeQuant support prim::If
* #35141 [quant][graphmode][refactor] getGeneralOpTensorInputIndexes -> getGeneralOpTensorInputs
* #35135 [quant][graphmode][refactor] swapDeQuant takes block as arugment
* #35077 [quant][graphmode] Fold quantized prepacking ops
* #35073 [quant] Make conv2d_prepack and linear_prepack pure
* #34855 [quant][graphmode] quantization support for prim::CallFunction
* **#34346 [quant][graphmode] Add quantization support for aten::cat**

Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20589145](https://our.internmc.facebook.com/intern/diff/D20589145)

raghuramank100 commented on torch/csrc/jit/passes/quantization_patterns.h:
```diff
@@ -127,6 +127,18 @@ graph(%packed_params, %a_quant, %r_scale, %r_zero_point, %r_dtype):
         %r = quantized::linear(%a_quant, %packed_params, %r_scale, %r_zero_point)
         return (%r) )";
 
+  std::string cat = R"(
```
This needs to work with a list of tensors and for lists of arbitrary length (known). Please add a test.

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar reviewed (approved):
LGTM

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20589145](https://our.internmc.facebook.com/intern/diff/D20589145)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20589145](https://our.internmc.facebook.com/intern/diff/D20589145)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20589145](https://our.internmc.facebook.com/intern/diff/D20589145)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20589145](https://our.internmc.facebook.com/intern/diff/D20589145)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization support for aten::cat"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20589145](https://our.internmc.facebook.com/intern/diff/D20589145)

[ghstack-poisoned]

albanD commented:
Reverting this as the newly introduced test is failing on master for macos build: https://app.circleci.com/pipelines/github/pytorch/pytorch/145635/workflows/e811c5de-9af3-4709-9f98-8b88e515b4e5/jobs/4934900
